setup.py

```diff
@@ -1,5 +1,31 @@
 from setuptools import setup, find_packages
 
+
+def union(*deps):
+    return list(set().union(*[extras[dep] for dep in deps]))
+
+extras = {
+    "envs": [
+        "atari_py~=0.2.0",            # atari environments
+        "box2d-py~=2.3.5",            # box2d environments
+        "pybullet>=3.0.6",            # continuous environments
+        "Pillow",                     # rendering library
+    ],
+    "test": [
+        "pylint>=2.6.0",              # code quality tool
+        "torch-testing>=0.0.2",       # pytorch assertion library
+    ],
+    "docs": [
+        "sphinx>=3.2.1",              # documentation library
+        "sphinx-autobuild>=2020.9.1", # documentation live reload
+        "sphinx-rtd-theme>=0.5.0",    # documentation theme
+        "sphinx-automodapi>=0.13",    # autogenerate docs for modules
+    ]
+}
+
+extras["dev"] = union("envs", "test", "docs")
```

Comment:
"all" is what literally every single other python library calls this, not dev

Reply:
I couldn't find any clear examples of this outside of gym and PettingZoo. Are you aware of any other sources for this?Here's some random blog post recommending dev: https://snarky.ca/clarifying-pep-518/

My intuition on this is that something like `all` is really targeted at PyPi users where as `dev` is targeted at the maintainers of the repository. For example, gym[all] installs "all" of the available environments, but it doesn't install any dev dependencies. If they decided to add such dependencies, it would make sense to keep them separate so that users would still have an easy way of install "all" of the environments.

NodeJS follows this idea explicitly by separating "dependencies" from "dev dependencies": https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file